dart/lib/src/sentry_options.dart

```diff
@@ -138,8 +142,14 @@ class SentryOptions {
     this.environmentAttributes,
     this.compressPayload,
     this.httpClient,
-    this.clock,
-  });
+    dynamic clock,
+  }) {
+    if (clock == null) {
+      clock = getUtcDateTime;
+    } else {
+      _clock = (clock is ClockProvider ? clock : clock.get) as ClockProvider;
```

Comment:
@rxlabz can we deprecate one of them or is this due to Dart version?

Reply:
IMO we totally can type `clock` as ClockProvider.